bzlib.rs

```diff
@@ -1,39 +1,32 @@
+use crate::compress::BZ2_compressBlock;
+use crate::crctable::BZ2_crc32Table;
+use crate::decompress::BZ2_decompress;
+use crate::randtable::BZ2_rNums;
 use ::libc;
-use libc::{fprintf, FILE};
+use libc::FILE;
+use libc::{
+    exit, fclose, fdopen, ferror, fflush, fgetc, fopen, fread, free, fwrite, malloc, strcat,
+    strcmp, ungetc,
+};
+
 extern "C" {
-    fn malloc(_: libc::c_ulong) -> *mut libc::c_void;
-    fn free(_: *mut libc::c_void);
-    fn exit(_: libc::c_int) -> !;
     static stdin: *mut FILE;
     static stdout: *mut FILE;
     static stderr: *mut FILE;
-    fn fclose(__stream: *mut FILE) -> libc::c_int;
-    fn fflush(__stream: *mut FILE) -> libc::c_int;
-    fn fopen(_: *const libc::c_char, _: *const libc::c_char) -> *mut FILE;
-    fn fdopen(__fd: libc::c_int, __modes: *const libc::c_char) -> *mut FILE;
-    fn fgetc(__stream: *mut FILE) -> libc::c_int;
-    fn ungetc(__c: libc::c_int, __stream: *mut FILE) -> libc::c_int;
-    fn fread(
-        _: *mut libc::c_void,
-        _: libc::c_ulong,
-        _: libc::c_ulong,
-        _: *mut FILE,
-    ) -> libc::c_ulong;
-    fn fwrite(
-        _: *const libc::c_void,
-        _: libc::c_ulong,
-        _: libc::c_ulong,
-        _: *mut FILE,
-    ) -> libc::c_ulong;
-    fn ferror(__stream: *mut FILE) -> libc::c_int;
     fn __ctype_b_loc() -> *mut *const libc::c_ushort;
-    fn strcat(_: *mut libc::c_char, _: *const libc::c_char) -> *mut libc::c_char;
-    fn strcmp(_: *const libc::c_char, _: *const libc::c_char) -> libc::c_int;
-    static mut BZ2_rNums: [Int32; 512];
-    static mut BZ2_crc32Table: [UInt32; 256];
-    fn BZ2_compressBlock(_: *mut EState, _: Bool);
-    fn BZ2_decompress(_: *mut DState) -> Int32;
 }
+
+macro_rules! version {
+    () => {
+        "1.1.0"
```

Comment:
```suggestion
        env!("CARGO_PKG_VERSION")
```

if you want to use the version from Cargo.toml. Or maybe

```suggestion
        concat!("1.1.0-rs", env!("CARGO_PKG_VERSION"))
```

to indicate that it is based on bzip 1.1.0 while still independently providing a version for the rust port?

Reply:
yeah, we do something clever here in zlib-rs. Let's come back to that later

Reply:
tracking issue https://github.com/trifectatechfoundation/libbzip2-rs/issues/2